apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/source/RdaVersion.java

```diff
@@ -1,8 +1,21 @@
 package gov.cms.bfd.pipeline.rda.grpc.source;
 
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 /** A utility class for comparing version numbers. */
 public class RdaVersion {
 
+  private static final String TYPE_GROUP = "type";
+  private static final String MAJOR_GROUP = "major";
+  private static final String MINOR_GROUP = "minor";
+  private static final String PATCH_GROUP = "patch";
+  private static final Pattern VERSION_PATTERN =
+      Pattern.compile(
+          String.format(
+              "^(?<%s>[~^])?(?<%s>\\d+)\\.(?<%s>\\d+)\\.(?<%s>\\d+)$",
+              TYPE_GROUP, MAJOR_GROUP, MINOR_GROUP, PATCH_GROUP));
```

Comment:
do we need to be concerned w/ "SNAPSHOT" as well in this regex ??

https://cmsgov.slack.com/archives/C01KG2D14TH/p1677265519295269?thread_ts=1677264955.121689&cid=C01KG2D14TH

Reply:
A snapshot version in production would be highly undesired, having the process fail to parse it would seem sufficient.